clang/lib/Driver/ToolChains/Flang.cpp

```diff
@@ -777,6 +777,15 @@ void Flang::ConstructJob(Compilation &C, const JobAction &JA,
 
   addFortranDialectOptions(Args, CmdArgs);
 
+  // 'flang -E' always produces output that is suitable for use as fixed form
+  // Fortran. However it is only valid free form source if the original is also
+  // free form.
+  if (InputType == types::TY_PP_Fortran &&
+      !Args.hasArg(options::OPT_ffixed_form) &&
+      !Args.hasArg(options::OPT_ffree_form)) {
```

Comment:
This should capture the right-most instance of `-ffixed-form`/`-ffree-form`, which is exactly what you want here:
```cpp
            Args.getLastArg(clang::driver::options::OPT_ffixed_form,
                            clang::driver::options::OPT_ffree_form))
```

Reply:
Updated to use getLastArg

Reply:
I'm not seeing `getLastArg` here on the github review.. Is this github being weird?

Reply:
> I'm not seeing `getLastArg` here on the github review.. Is this github being weird?

Oops. **Really** updated to use getLastArg now.